docs/source/transaction_family_specifications/validator_registry_transaction_family.rst

```diff
@@ -0,0 +1,257 @@
+namespace for the validator registry will be the first 6 characters of the
+sha256  hash of the string “validator_registry”, which is “6a4372” For example,
+the validator signup info of a validator with the id “validator_id” could be
+set like this:
+
+.. code-block:: pycon
+
+  >>>“6a4372” + hashlib.sha256('validator_id'.encode("utf-8")).hexdigest()
+  '6a43722aee5b550a3cbd1595f4de10049ee805bc035b5e232dfacfc31cc6275170b30d'
+
+The map of current the current registered validator_id should be stored at the
+ following address:
+
+
+.. code-block:: pycon
+
+  >>>“6a4372” + hashlib.sha256('validator_map'.encode()).hexdigest()
+  '6a437247a1c12c0fb03aa6e242e6ce988d1cdc7fcc8c2a62ab3ab1202325d7d677e84c'
+
+Transaction Payload
+===================
+Validator registry transaction family payloads are defined by the following
+protocol buffers code:
+
+.. code-block:: protobuf
+
+  message ValidatorRegistryPayload {
+    // The action that the transaction processor will take. Currently this
+    // is only “register”, but could include other actions in the futures
+    // such as “revoke”
+  	string verb = 1;
+
+    // The name of the endpoint. It should be a human readable name.
+  	string name = 2;
+
+
+  	// validator's public key (currently using signer_pubkey as this is
+    // stored in the transaction header)
+      string id = 3;
+
+    // This is the protocol buffer described above.
+    SignUpInfo signup_info = 4;
+
+    // The current chain head’s block num. This is used as a form of time
+    // to help with the enforcement of the C and K tests
+    int32 block_num = 5:
+
+  }
+
+
+
+Transaction Header
+==================
+
+Inputs and Outputs
+------------------
+
+The inputs for validator registry family transactions must include:
+
+* the address of *validator_id*
+* the address of *validator_map*
+
+The outputs for validator registry family transactions must include:
+
+* the address of *validator_id*
+* the address of *validator_map*
+
+
+Dependencies
+------------
+
+None
+
+Family
+------
+- family_name: "sawtooth_validator"
+- family_version: "1.0"
+
+
+Encoding
+--------
+
+The encoding field must be set to "application/protobuf".
+
+
+Execution
+=========
+
+Untrusted python code, that is a part of the transaction processor, will verify
+the attestation verification report for the signup information. It is important to
+note, that the IAS keys will need to be on the block chain and it will need to
+be set on configuration. This will allow both the simulator logic and real
+SGX logic to be the same.
+
+Before the validator_registry transaction can be sent the SignUpInfo needs to be
+created by the poet_consensus and it  will need to request the current
+chain_head’s block number from the journal.
+
+The journal will need a new method to retrieve this information.
+
+If the validator_name does not match syntactic requirements, the transaction is
+invalid. The current requirement is that the validator_name is 64 characters
+or less.
```

Comment:
And presumably non-empty???  Although really that is just for human consumption, so maybe that doesn't matter.